packages/fuel-gauge/src/contract-factory.test.ts

```diff
@@ -85,16 +85,33 @@ describe('Contract Factory', () => {
     });
     expect(transactionResult.gasUsed.toNumber()).toBeGreaterThan(0);
 
-    const { callResult } = await contact.functions.increment_counter(1).dryRun();
+    const { callResult } = await contract.functions.increment_counter(1).dryRun();
     expect(callResult).toMatchObject({
       receipts: expect.arrayContaining([expect.any(Object)]),
     });
   });
 
+  it('can deploy a contract asynchronously without waiting for the TX to be processed', async () => {
+    const factory = await createContractFactory();
+
+    const { transactionResponse, contract } = await factory.deployContract();
+
```

Comment:
@petertonysmith94 An error will be thrown because the contract was not deployed yet, therefore, it does not exist.

Reply:
Worthy of mentioning is that if the `awaitExecution = true` argument is passed, then there is no need to call `waitForResult`. Whoever uses the non-blocking `awaitExecution = false` should accept the nature of blockchain and do a `waitForResult` before running contract calls.

Reply:
I understand the rationale and that this is a delicate issue.

Still, I suspect this error case can be counterintuitive.

Reply:
> Still, I suspect this error case can be counterintuitive.

If we want to still _not_ fail a contract call and not have this error case, we could couple deployment to contract calls and wait for deployment to finish before running the actual contract call:
```ts
// mock internals of a contract call
await this.deploymentTransactionResponse?.waitForResult(); 
return call();
```

Then we can just explain in the documentation that, for `awaitExecution=false`, the first contract call might take a bit longer because the contract might not have been deployed yet.

Reply:
@Torres-ssf what would be the error type thrown if one calls the contract before it's deployed? I think a better path would be to document that error so that consumers can error handle appropriately and retry when necessary.

Reply:
We could consider [splitting the functionality](https://github.com/FuelLabs/fuels-ts/pull/2597#discussion_r1658507262).

Reply:
https://github.com/FuelLabs/fuels-ts/pull/2597/commits/c0b06bbc64a49daaae493b3a83cdfc78b56de7df